File: kotlin/src/main/kotlin/com/di/kotlin/di/solo/UserService.kt
package com.di.kotlin.di.solo

import com.di.kotlin.ILogger
import com.di.kotlin.IUserRepository
import com.di.kotlin.IUserService
import com.di.kotlin.IValidationsService
import com.di.kotlin.model.User

class UserService(private val userRepository: IUserRepository,
                  private val logger: ILogger,
                  private val validationsService: IValidationsService) : IUserService {

    override fun create(nombre: String, apellido: String): User {
        val valid = this.validationsService.validateIdentidad(nombre, apellido)
        if (!valid) {
            throw Exception("El nombre o el apellido no son validos")
        }
        val user = this.userRepository.createUser(nombre, apellido)
        this.logger.log("User created: $user")
        return user
    }

    override fun findById(id: Long) : User? {
        return this.userRepository.findById(id)
    }
}

File: nodejs/src/sinDependencyInjection/UserService.js
import {UserRepository} from "./UserRepository";
import {Logger} from "./Logger";
import {ValidationsService} from "./ValidationsService";

export class UserService {
    userRepository = new UserRepository();
    logger = new Logger();
    validationsService = new ValidationsService();

    create(nombre, apellido) {
        const valid = this.validationsService.validateIdentidad(nombre, apellido);
        if (!valid) {
            throw Error("El nombre o el apellido no son validos");
        }
        const user = this.userRepository.create(nombre, apellido);
        this.logger.log(`User created: ${JSON.stringify(user,null,2)}`);
        return user
    }

    findById(id) {
        return this.userRepository.findById(id);
    }
}